### PR TITLE
[Issue 92] performance testing initial support

### PR DESCRIPTION
**Pull request by andkononykhin**
- fixes unit tests to be compatible with DEX (version <040) Android format so they can be run on Android API level <30
- adds naive performance testing (looping) for local (JVM) and Android testing
- adds performance testing basing on [Java Microbenchmark Harness (JMH)](https://github.com/openjdk/jmh) framework
- adds docs for the topic

addresses https://github.com/sicpa-dlab/didcomm-gemini/issues/92

### CODE REVIEW

File: lib/src/testCommon/kotlin/org/dif/fixtures/BenchCommon.kt [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            "'${this.opName}'", this.timeInMs, operations, this.thrpt, this.avrg
+        )
+    }
+}
+
+class BenchCommon {
+    companion object {
+        val didComm_def: DIDComm = DIDComm(DIDDocResolverMock(), AliceSecretResolverMock())
+
+        val packSignedRes = didComm_def.packSigned(
```
Comment (ashcherbakov) [inline lib/src/testCommon/kotlin/org/dif/fixtures/BenchCommon.kt:32]: Camel case is usually used for Java/Kotlin vars and methods

File: benchmark/src/androidTest/kotlin/org/dif/DIDCommBenchAndroidNaive.kt [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @Test
+    fun bench_pack_signed() {
+        // TODO resolve fun name dynamically
+        measure(
+            BenchCommon::pack_signed,
+            aFunName = "pack_signed"
+        )
+        /*
```
Comment (ashcherbakov) [inline benchmark/src/androidTest/kotlin/org/dif/DIDCommBenchAndroidNaive.kt:34]: Can we remove that commented lines?

**Review by ashcherbakov — APPROVED** [review]
LGTM.
Just consider following Java/Kotlin style with naming (camel case).

File: benchmark/src/androidTest/kotlin/org/dif/DIDCommBenchAndroidNaive.kt (hunk 2 above)
Comment (andkononykhin) [reply benchmark/src/androidTest/kotlin/org/dif/DIDCommBenchAndroidNaive.kt:34]: the intention was to keep some traces of [Android benchmark native framework](https://developer.android.com/jetpack/androidx/releases/benchmark) usage for the future

File: lib/src/testCommon/kotlin/org/dif/fixtures/BenchCommon.kt (hunk 1 above)
Comment (andkononykhin) [reply lib/src/testCommon/kotlin/org/dif/fixtures/BenchCommon.kt]: fixed

**Review by ashcherbakov — APPROVED** [review]

**Pull request merged**